pvlib/tests/iotools/test_tmy.py

```diff
@@ -41,3 +45,31 @@ def test_read_tmy3_no_coerce_year():
 
 def test_read_tmy2():
     tmy.read_tmy2(TMY2_TESTFILE)
+
+
+def test_gh865_read_tmy3_feb_leapyear_hr24():
+    """correctly parse the 24th hour if the tmy3 file has a leap year in feb"""
+    data, meta = read_tmy3(TMY3_FEB_LEAPYEAR)
+    # just to be safe, make sure this _IS_ the Greensboro file
+    greensboro = {
+        'USAF': 723170,
+        'Name': '"GREENSBORO PIEDMONT TRIAD INT"',
+        'State': 'NC',
+        'TZ': -5.0,
+        'latitude': 36.1,
+        'longitude': -79.95,
+        'altitude': 273.0}
+    assert meta == greensboro
+    # February for Greensboro is 1996, a leap year, so check to make sure there
+    # aren't any rows in the output that contain Feb 29th
+    assert data['1996-02-29 00:00'].size == 0
```

Comment:
It's hard for me to track what's actually supposed to happen. Perhaps a test like

```python
assert data.index[67] = pd.Timestamp('1996-02-28 23:00')
assert data.index[68] = pd.Timestamp('1996-03-01 00:00')
```

assuming that's actually correct. And similar for the coerced year.